app/controllers/users_controller.rb

```diff
@@ -0,0 +1,57 @@
+class UsersController < ApplicationController
+
+	get '/login' do 
+		logged_in? ? (redirect '/tweets') : (erb :'/users/login')
+	end
+
+
+	post '/login' do
+		user = User.find_by(username: params[:username])
+			if user && user.authenticate(params[:password])
+				session[:user_id] = user.id
+				redirect '/tweets'
+			elsif !user
+				redirect '/signup'
+				flash[:message] = "Invalid user.  Please sign up."
+			end
+	end
+
+
+	get '/logout' do 
+		session.clear
+		success
+		redirect '/login'
+	end
+
+	
+	get '/signup' do 
+		if logged_in?
+			success
+			redirect '/tweets'
+		else
+			erb :'/users/create_user'
+		end
+	end
+
+
+	post '/signup' do 
+		@user = User.create(params) if filled_out(params)
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.